tests/Aspire.Hosting.Testing.Tests/TestingBuilderTests.cs

```diff
@@ -13,6 +16,58 @@ namespace Aspire.Hosting.Testing.Tests;
 
 public class TestingBuilderTests
 {
+    [Fact]
+    [RequiresDocker]
+    public async Task CanLoadFromDirectoryOutsideOfAppContextBaseDirectory()
+    {
+        // This test depends on the TestProject.AppHost not being in `AppContext.BaseDirectory` for the tests assembly.
+        var unexpectedAppHostFiles = Directory.GetFiles(AppContext.BaseDirectory, "TestProject.AppHost.*");
+        if (unexpectedAppHostFiles.Length > 0)
+        {
+            // The test requires that the TestProject.AppHost* files not be present in the test directory
+            // This is a defensive check to ensure that the test is not run in an unexpected environment due
+            // to build changes
+            throw new InvalidOperationException($"Found unexpected AppHost files in {AppContext.BaseDirectory}: {string.Join(", ", unexpectedAppHostFiles)}");
+        }
+
+        var testProjectAssemblyPath = Directory.GetFiles(
+            Path.Combine(MSBuildUtils.GetRepoRoot(), "artifacts", "bin", "TestProject.AppHost"),
+            "TestProject.AppHost.dll",
+            SearchOption.AllDirectories).FirstOrDefault();
+
+        Assert.True(File.Exists(testProjectAssemblyPath), $"TestProject.AppHost.dll not found at {testProjectAssemblyPath}.");
+
+        var appHostAssembly = Assembly.LoadFrom(Path.Combine(AppContext.BaseDirectory, testProjectAssemblyPath));
+        var appHostType = appHostAssembly.GetTypes().FirstOrDefault(t => t.Name.EndsWith("_AppHost"))
+            ?? throw new InvalidOperationException("Generated AppHost type not found.");
+
+        TestResourceNames resourcesToSkip = ~TestResourceNames.redis;
+        var appHost = await DistributedApplicationTestingBuilder.CreateAsync(appHostType, ["--skip-resources", resourcesToSkip.ToCSVString()]);
+        await using var app = await appHost.BuildAsync();
+        await app.StartAsync();
+
+        // Sanity check that the app is running as expected
+        // Get an endpoint from a resource
+        var serviceAHttpEndpoint = app.GetEndpoint("servicea", "http");
+        Assert.NotNull(serviceAHttpEndpoint);
+        Assert.True(serviceAHttpEndpoint.Host.Length > 0);
+    }
+
+    [Fact]
+    public async Task ThrowsForAssemblyWithoutAnEntrypoint()
+    {
+        var classLibraryAssemblyPath = Directory.GetFiles(
+            AppContext.BaseDirectory,
+            "ClassLibrary.dll",
+            SearchOption.AllDirectories).Single();
+
+        var appHostAssembly = Assembly.LoadFrom(Path.Combine(AppContext.BaseDirectory, classLibraryAssemblyPath));
+        var appHostType = appHostAssembly.GetTypes().Single(t => t.FullName == "ClassLibrary.Class1");
```

Comment:
Instead of a whole new assembly, could we just use an existing assembly? Like `Aspire.Components.Common.Tests`?

Reply:
`Aspire.Components.Common.Tests` gets an auto-generated `.entrypoint`:
```
.custom instance void [Microsoft.VisualStudio.TestPlatform.ObjectModel]Microsoft.VisualStudio.TestPlatform.TestSDKAutoGeneratedCode::.ctor()
```
I randomly used `Microsoft.Extensions.Logging.ConsoleLoggerExtensions` instead.